src/metabase-report-executor/src/main/java/com/champtitles/metabasereportexecutor/executor/MetabaseClient.java

```diff
@@ -0,0 +1,245 @@
+     *
+     * @param baseUrl  URL of the Metabase server
+     * @param email    login email
+     * @param password login password
+     */
+    public MetabaseClient(String baseUrl, String email, String password) {
+        this.baseUrl = baseUrl;
+        this.email = email;
+        this.password = password;
+        httpClient = HttpClient.newBuilder().build();
+    }
+
+    /**
+     * Create a MetabaseClient
+     *
+     * @param baseUrl    URL of the Metabase server
+     * @param email      login email
+     * @param password   login password
+     * @param httpClient inject a HttpClient
+     */
+    public MetabaseClient(String baseUrl, String email, String password, HttpClient httpClient) {
+        this.baseUrl = baseUrl;
+        this.email = email;
+        this.password = password;
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Query the metabase server for session properties. This will include the setup token which will only
+     * be present if the server has not yet been initialized.
+     * <p>
+     * metabase.com/docs/latest/api/session#get-apisessionproperties
+     *
+     * @return metabase session properties
+     */
+    public SessionPropertiesResponse getSessionProperties() {
+        HttpRequest httpRequest = HttpRequest
+                .newBuilder()
+                .uri(createUri("/api/session/properties"))
+                .GET()
+                .build();
+        String response = sendHttpRequest(httpRequest, 200);
+
+        try {
+            return objectMapper.readValue(response, SessionPropertiesResponse.class);
+        } catch (JsonProcessingException e) {
+            logger.error("failed to parse session properties response: {}", response);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Complete the initial setup of the metabase server by creating a user and setting basic properties
+     * <p>
+     * metabase.com/docs/latest/api/setup#post-apisetup
+     *
+     * @param setupToken setup token obtained by querying the session properties API
+     */
+    public void completeInitialSetup(String setupToken) {
+        SetupRequest setupRequest = new SetupRequest(setupToken, email, password);
+
+        HttpRequest httpRequest = HttpRequest
+                .newBuilder()
+                .uri(createUri("/api/setup"))
+                .header("Content-Type", "application/json")
+                .POST(createBody(setupRequest))
+                .build();
+        String response = sendHttpRequest(httpRequest, 200);
+        logger.info("setup response: {}", response);
+    }
+
+    /**
+     * Log in to metabase with the configured email and password and store the session id
+     * <p>
+     * metabase.com/docs/latest/api/session#post-apisession
+     *
+     * @return session id (cookie) to be used for subsequent requests
+     */
+    public String loginAndGetSession() {
+        SessionRequest sessionRequest = new SessionRequest(email, password);
```

Comment:
@dcrough-ownum the signature of this method may be confusing since it returns a String but it does store the session id as an instance variable so it can be reused. 
`sessionId = sessionResponse.id();`

I am also returning the session id as a convenience.